### PR TITLE
Change ashwalkers to have custom lungs

### PR DESCRIPTION
**Pull request by coiax**
Ashwalkers now have lungs that are specially adapted to Lavaland
environments. However, that specialisation comes at a cost at sometimes
not being able to operate in normal human air conditions.

- Ashwalker lungs take a gas sample of the default lavaland atmosphere
  on initialisation in order to customise themselves
- Currently, excess O2/N2 has no functional drawback, so no maximum
  value is set on the lungs. If Lavaland rolls a higher N2/O2 mole count
  than human air, they will still struggle however.
- Ashwalkers are no longer NOBREATH, bringing them closer
  to regular lizards.

:cl: coiax
balance: Ashwalkers now have lungs adapted for the atmosphere of Indecipheres,
and now breathe. However, if the Indecipheres
atmosphere has miasma, they are immune to it.
/:cl:

### CODE REVIEW

**Comment by dragomagol** [comment]
Shouldn't this have a changelog? Seems like something important for ghosts playing ashwalkers to know.

**Comment by rasepretep** [comment]
Would this make it possible to get an Ashwalker (not ghost-spawned) corpse from a legion, epi it and get those lungs transplanted to gain that benefit? Alternatively, could this open a path to OOC character customization, allowing lizard players to not only pick regular/digitigrade legs, but regular/ashwalker lungs as well?

**Comment by coiax** [comment]
> 
> 
> Shouldn't this have a changelog? Seems like something important for ghosts playing ashwalkers to know.

You're completely right.

**Comment by D4C-420** [comment]
Don't they have VIRUSIMMUNE because they don't have any reasonable way to clean blood and gibs and as a result will get sick from walking around in it? I mean it should be miners and fauna killing ash walkers not random illness they can't cure.

**Comment by coiax** [comment]
As far as I understand, the sickness from gibs and blood comes from miasma buildup, which won't happen in the lavaland atmosphere, because it's constantly being reset to the normal air values.

And if miasma is normally part of the air, then they're immune to it.

**Comment by MrMelbert** [comment]
as of #57957 , gibs / dead people no longer spit out miasma and instead try to infect you with diseases directly. Ashwalkers were given virus immunity a month or so ago because their tendril is often coated in gibs and dead bodies, which very often infected them with nasty viruses.

**Comment by coiax** [comment]
> 
> 
> as of #57957 , gibs / dead people no longer spit out miasma and instead try to infect you with diseases directly. Ashwalkers were given virus immunity a month or so ago because their tendril is often coated in gibs and dead bodies, which very often infected them with nasty viruses they had no means of curing or preventing.

Did not know this, I will restore their virus immunity.

> Side question: If you take ash-walker mutation toxin on the station, will your lungs be functionally the same as normal lungs?

The lungs are "prototyped" on the ideal lavaland gas that the air controller stores, ie. it's not on "first breath" or anything. Ashwalkers (or more specifically, blackened frilled lungs) created anywhere will be based on lavaland atmosphere.
Hilariously, even if the map is icebox.

**Comment by MrMelbert** [comment]
> The lungs are "prototyped" on the ideal lavaland gas 

Oh well, seems like gone are the days of sipping ashwalker mutation toxin in botany for nobreath and virus immunity. Still a cool change, though.

**Comment by ATH1909** [comment]
time to turn people into ash walkers as a changeling to make them suffocate 😎

File: code/modules/surgery/organs/lungs.dm [hunk 1]
```diff
@@ -562,3 +564,54 @@
...
+	var/carbon_dioxide_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/carbon_dioxide][MOLES])
+	var/bz_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/bz][MOLES])
+	var/miasma_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/miasma][MOLES])
+
+	// Normal oxygen is 21 kPa partial pressure, but SS13 humans can tolerate down
+	// to 16 kPa. So it follows that ashwalkers, as humanoids, follow the same rules.
+	var/TOLERANCE = 5
+	safe_oxygen_min = max(0, oxygen_pp - TOLERANCE)
+	safe_nitro_min = max(0, nitrogen_pp - TOLERANCE)
+	safe_toxins_min = max(0, plasma_pp - TOLERANCE)
```
Comment (ATH1909) [inline code/modules/surgery/organs/lungs.dm:604]: 👀

Comment (coiax) [reply code/modules/surgery/organs/lungs.dm:604]: You're going to have to be more specific.

Comment (Mothblocks) [reply code/modules/surgery/organs/lungs.dm:604]: he never is

File: code/modules/surgery/organs/lungs.dm [hunk 2]
```diff
@@ -562,3 +564,54 @@
...
+	var/oxygen_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/oxygen][MOLES])
+	var/nitrogen_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/nitrogen][MOLES])
+	var/plasma_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/plasma][MOLES])
+	var/carbon_dioxide_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/carbon_dioxide][MOLES])
+	var/bz_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/bz][MOLES])
+	var/miasma_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/miasma][MOLES])
+
+	// Normal oxygen is 21 kPa partial pressure, but SS13 humans can tolerate down
+	// to 16 kPa. So it follows that ashwalkers, as humanoids, follow the same rules.
+	var/TOLERANCE = 5
```
Comment (Mothblocks) [inline code/modules/surgery/organs/lungs.dm:601]: Can you make this a define? I don't like SCREAMING_VARIABLES when they're just normal variables, confusing.

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm [hunk 3]
```diff
@@ -116,12 +116,11 @@ Lizard subspecies: ASHWALKERS
 	id = "ashlizard"
 	limbs_id = "lizard"
 	species_traits = list(MUTCOLORS,EYECOLOR,LIPS,DIGITIGRADE,HAS_FLESH,HAS_BONE)
+	mutantlungs = /obj/item/organ/lungs/ashwalker
 	inherent_traits = list(
 		TRAIT_ADVANCEDTOOLUSER,
 		TRAIT_CAN_STRIP,
 		TRAIT_CHUNKYFINGERS,
-		TRAIT_NOBREATH,
-		TRAIT_VIRUSIMMUNE,
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:124]: as you said, restore this

File: code/modules/surgery/organs/lungs.dm [hunk 4]
```diff
@@ -562,3 +564,54 @@
 		COOLDOWN_START(src, severe_cooldown, 30 SECONDS)
 	if(prob(emp_vulnerability/severity)) //Chance of permanent effects
 		organ_flags |= ORGAN_SYNTHETIC_EMP //Starts organ faliure - gonna need replacing soon.
+
+
+/obj/item/organ/lungs/ashwalker
+	name = "blackened frilled lungs" // blackened from necropolis exposure
+	desc = "Exposure to the necropolis has mutated these lungs to breathe the air of the lava-covered moon below."
+	color = "#964b00" // TODO get proper sprite instead of BROWN
```
Comment (Mothblocks) [inline code/modules/surgery/organs/lungs.dm:572]: is this TODO now

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/lungs.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/surgery/organs/lungs.dm:604]: this is your only warning from me--stop making reviews like this

File: code/modules/surgery/organs/lungs.dm [hunk 5]
```diff
@@ -562,3 +564,54 @@
...
+
+/obj/item/organ/lungs/ashwalker/Initialize()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/surgery/organs/lungs.dm:574]: initialize has some arguments

**Comment by coiax** [comment]
Help the stale bot is bullying me

**Comment by Mothblocks** [comment]
its still drafted

**Comment by coiax** [comment]
@Mothblocks I'm in the testing stage.

**Pull request merged**